electrum/gui/qt/recovery_list.py

```diff
@@ -131,103 +126,30 @@ def __init__(self, parent, wallet: Abstract_Wallet, config):
-                super().initStyleOption(option, index)
-
-        self.recovery_privkey_line = CompletionTextEdit()
-        self.recovery_privkey_line.setTabChangesFocus(False)
-        self.recovery_privkey_line.textChanged.connect(self.on_priv_key_line_edit)
-
-        delegate = CompleterDelegate(self.recovery_privkey_line)
-        self.completer = QCompleter(self.wordlist)
-        self.completer.popup().setItemDelegate(delegate)
-        self.recovery_privkey_line.set_completer(self.completer)
-        # size hint other
-        height = self.recovery_address_line.sizeHint().height()
-        self.recovery_privkey_line.setMaximumHeight(height)
-        ###
-
-        grid_layout.addWidget(self.recovery_privkey_line, 1, 1)
-        # Row 3
-        button = QPushButton(_('Recover'))
-        button.clicked.connect(self.recover_action)
-        # if line edit with suggestions size change 3rd argument needs to be adjusted
-        grid_layout.addWidget(button, 2, 0, 1, 3)
-        ###
-
-        self.main_layout.addLayout(grid_layout)
-        self.setLayout(self.main_layout)
 
     def on_priv_key_line_edit(self):
-        for word in self.get_seed()[:-1]:
+        line = self.sender()
+        for word in line.text().split()[:-1]:
             if word not in self.wordlist:
-                self.recovery_privkey_line.disable_suggestions()
+                line.disable_suggestions()
                 return
-        self.recovery_privkey_line.enable_suggestions()
+        line.enable_suggestions()
 
-    def get_seed(self):
+    def get_recovery_seed(self):
         text = self.recovery_privkey_line.text()
         return text.split()
 
-    @staticmethod
-    def generate_encryption_key(salt: bytes, password: bytes) -> [bytes, int]:
-        kdf = PBKDF2HMAC(
-            algorithm=hashes.SHA256(),
-            length=32,
-            salt=salt,
-            iterations=100000,
-            backend=default_backend()
-        )
-        key = kdf.derive(password)
-        return key
-
     def _get_recovery_keypair(self):
-        recovery_pubkey = self.wallet.storage.get('recovery_pubkey')
-
-        seed = self.get_seed()
-        index_cache = [self.wordlist.index(word) for word in seed]
-
-        payload = 0
-        for i, b in enumerate(reversed(index_cache)):
-            payload = payload | (b << (i * 11))
-
-        payload = payload.to_bytes(17, 'big')
-        checksum = payload[:1]
-        entropy = payload[1:]
-
-        sha256sum = sha256(entropy).digest()
-        if sha256sum.hex()[0] != checksum.hex()[1]:
-            raise ValueError(_('Checksum miss matched'))
-
-        private_key = self.generate_encryption_key(entropy, entropy)
-
-        return {recovery_pubkey: (private_key, True)}
+        stored_recovery_pubkey = self.wallet.storage.get('recovery_pubkey')
+        seed = self.get_recovery_seed()
+        if not short_mnemonic.is_valid(seed):
+            raise ValueError(_("Invalid recovery TX seed"))
```

Comment:
The exception is caught in the caller function, and the message is displayed to the user, so I think it should be translated. Of course it would be good to improve the exception handling, so that this translation is not needed.

Reply:
ok :+1: